lmd/peer.go

```diff
@@ -193,6 +193,31 @@ func (p *Peer) Stop() {
 	close(p.shutdownChannel)
 }
 
+func (p *Peer) countFromServer(name string, queryCondition string) (count int) {
+	err, res := p.QueryString("GET "+name+"\nStats: "+queryCondition+"\n\n") 
+	count, e := strconv.Atoi(fmt.Sprintf("%s",res))
+	if e!=nil {
+		log.Debugf("problems getting count for %s (err: %s, res: %s): %s\n", name, err, res, e)
+		count = -1
+	} 
+	return
+}
+
+func (p *Peer) hasChanged() (changed bool) {
+	changed = false
+	tablenames := []string {"commands", "comments", "contactgroups", "contacts", "downtimes", "hostgroups", "hosts", "servicegroups", "services", "timeperiods"}
+	for _, name := range tablenames {
+		counter := p.countFromServer(name, "name >= .*")
+		if counter<0  {
+			counter = p.countFromServer(name, "host_name >= .*")
```

Comment:
this is quite ugly and will probably result in error messages in the livestatus.log for each check.